src/braket/experimental/autoqasm/program/program.py

```diff
@@ -509,10 +510,23 @@ def add_io_declarations(self) -> None:
                 root_oqpy_program.undeclared_vars[parameter.name]._needs_declaration = True
             else:
                 root_oqpy_program._add_var(parameter)
+
         for parameter_name, parameter in self._output_parameters.items():
             # Before adding the output variable to the program, remove any existing reference
-            root_oqpy_program.undeclared_vars.pop(parameter_name, None)
-            root_oqpy_program.declared_vars.pop(parameter_name, None)
+            popped_undeclared = root_oqpy_program.undeclared_vars.pop(parameter_name, None)
+            popped_declared = root_oqpy_program.declared_vars.pop(parameter_name, None)
+
+            popped = popped_undeclared if popped_undeclared is not None else popped_declared
```

Comment:
Because we know we won't see one parameter in both lists, right?

Reply:
That should certainly be the case. But it's a good point, I'll add an assert here to verify that we never have both.